Chess/src/Board.cpp

```diff
@@ -0,0 +1,134 @@
+#include "Board.h"
+#include "Rook.h"
+#include "King.h"
+#include "Bishop.h"
+#include "Queen.h"
+
+Board::Board() : _turn(true) // White starts
+{
+	//white pieces
+	_pieces.push_back(std::make_unique<Rook>(true, 0, 0, this));
+	_pieces.push_back(std::make_unique<Bishop>(true, 0, 2, this));
+	_pieces.push_back(std::make_unique<King>(true, 0, 3, this));
+	_pieces.push_back(std::make_unique<Queen>(true, 0, 4, this));
+	_pieces.push_back(std::make_unique<Bishop>(true, 0, 5, this));
+	_pieces.push_back(std::make_unique<Rook>(true, 0, 7, this));
+
+	//black pieces
+	_pieces.push_back(std::make_unique<Rook>(false, 7, 0, this));
+	_pieces.push_back(std::make_unique<Bishop>(false, 7, 2, this));
+	_pieces.push_back(std::make_unique<King>(false, 7, 3, this));
+	_pieces.push_back(std::make_unique<Queen>(false, 7, 4, this));
+	_pieces.push_back(std::make_unique<Bishop>(false, 7, 5, this));
+	_pieces.push_back(std::make_unique<Rook>(false, 7, 7, this));
+	
+	_whiteKing = (King*)getPiece(0, 3);
+	_blackKing = (King*)getPiece(7, 3);
+}
+
+Board::~Board()
+{
+	delete _whiteKing;
+	delete _blackKing;
+}
+
+bool Board::isEmpty(int row, int col)
+{
+	return getPiece(row, col) ? false : true;
```

Comment:
Instead of return getPiece(row, col) ? false : true;, you can directly return the result of !getPiece(row, col);.